### PR TITLE
Fix adding empty char set atoms in Regexp TreeBuilder of `:fromString()` method

### LINKED ISSUE #39458: [Bug]: Regex causes a CCE in Runtime
### Description

Consider the following example in ballerina.

```
import ballerina/lang.regexp;

public function main() returns error? {
    regexp:RegExp rA = check regexp:fromString("[a-z]"); // ^[^a-zA-Z0-9]"
    string phrase = "TLearn/ Ballerina^ in";
    _ = check (trap rA.find(phrase, 4));
}
```

When executed, it would cause the following runtime exception:

```
error: java.lang.ClassCastException {"message":"class io.ballerina.runtime.internal.values.RegExpCharSet cannot be cast to class java.lang.String (io.ballerina.runtime.internal.values.RegExpCharSet is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')"}
```

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #39458

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pcnfernando** [comment]
@sanjana can you check on the build and request reviewers?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/RegExpFactory.java [hunk 1]
```diff
@@ -192,7 +192,10 @@ private static RegExpAtom translateCharacterClass(RegExpCharacterClass charClass
                 range.setRhsCharSetAom(translateCharInCharacterClass(range.getRhsCharSetAtom()));
                 continue;
             }
-            charAtoms[i] = translateCharInCharacterClass((String) charAtom);
+
```
Comment (pcnfernando) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/RegExpFactory.java:195]:
```suggestion
```

File: langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java [hunk 2]
```diff
@@ -97,6 +115,15 @@ public static BArray findAllGroups(BRegexpValue regExp, BString str, long startI
         return groupArray;
     }
 
+    private static Matcher getMatcher(BRegexpValue regExp, BString str) {
```
Comment (pcnfernando) [inline langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java:118]: Why do we have this here? Can't we use RegexpUtil's version

Comment (sanjana) [reply langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java]: Was handled in #39444. 
Re-synced from the master and verified that this is fixe.

**Review by mohanvive — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**